在 dotnet 5 构建失败

```
/home/lindexi/.nuget/packages/dotnetcampus.sourcefusion/0.9.6-alpha/build/dotnetCampus.SourceFusion.targets(84,5): error MSB3073: The command "dotnet "/home/lindexi/.nuget/packages/dotnetcampus.sourcefusion/0.9.6-alpha/build/../tools/netcoreapp3.1/codet.dll" "/home/lindexi/workspace/BUILD_DOT_NET_IMAGE_TMP/LindexiOpenXml.Core" -t obj/release/SourceFusion.Tools/ -c obj/release/SourceFusion.GeneratedCodes/ -p "obj/release/SourceFusion.Tools/CommandArgs.txt" -s "TRACE;RELEASE;NETCOREAPP;NETCOREAPP3_1;;NETCOREAPP1_0_OR_GREATER;NETCOREAPP1_1_OR_GREATER;NETCOREAPP2_0_OR_GREATER;NETCOREAPP2_1_OR_GREATER;NETCOREAPP2_2_OR_GREATER;NETCOREAPP3_0_OR_GREATER;NETCOREAPP3_1_OR_GREATER" --rebuild true" exited with code 150.
```

--- a/.github/workflows/dotnetcore.yml
+++ b/.github/workflows/dotnetcore.yml
@@ -16,4 +16,4 @@ jobs:
         dotnet-version: '6.0.x'
         include-prerelease: true
     - name: Build with dotnet
-      run: dotnet build --configuration Release
+      run: dotnet build --configuration Release -v n

--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -7,7 +7,7 @@
     <LangVersion>latest</LangVersion>
     <DevelopmentDependency>true</DevelopmentDependency>
     <PackageRequireLicenseAcceptance>false</PackageRequireLicenseAcceptance>
-    <Copyright>Copyright (c) 2018-2020 dotnet-campus</Copyright>
+    <Copyright>Copyright (c) 2018-2021 dotnet-campus</Copyright>
     <PackageProjectUrl>https://github.com/dotnet-campus/SourceFusion</PackageProjectUrl>
     <RepositoryUrl>https://github.com/dotnet-campus/SourceFusion.git</RepositoryUrl>
     <RepositoryType>git</RepositoryType>

--- a/SourceFusion.sln
+++ b/SourceFusion.sln
@@ -25,10 +25,13 @@ EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "samples", "samples", "{78213588-C3B8-4E86-A19B-73DECD1ABA7B}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "dotnetCampus.Telescope.Sample", "src\dotnetCampus.Telescope.Sample\dotnetCampus.Telescope.Sample.csproj", "{50AE25A2-DF1B-4DE9-A7BE-C406F6497376}"
+	ProjectSection(ProjectDependencies) = postProject
+		{84DA7EE9-B511-407E-A510-DF390C742F0D} = {84DA7EE9-B511-407E-A510-DF390C742F0D}
+	EndProjectSection
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "dotnetCampus.TelescopeTask", "src\dotnetCampus.TelescopeTask\dotnetCampus.TelescopeTask.csproj", "{09FF40A1-C13C-4D29-93B2-7A7C34D15694}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "dotnetCampus.Telescope.NuGet", "src\dotnetCampus.Telescope.NuGet\dotnetCampus.Telescope.NuGet.csproj", "{84DA7EE9-B511-407E-A510-DF390C742F0D}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "dotnetCampus.Telescope.NuGet", "src\dotnetCampus.Telescope.NuGet\dotnetCampus.Telescope.NuGet.csproj", "{84DA7EE9-B511-407E-A510-DF390C742F0D}"
 	ProjectSection(ProjectDependencies) = postProject
 		{09FF40A1-C13C-4D29-93B2-7A7C34D15694} = {09FF40A1-C13C-4D29-93B2-7A7C34D15694}
 		{556913DF-6532-4532-B514-2D1184ADC75B} = {556913DF-6532-4532-B514-2D1184ADC75B}

--- a/src/SourceFusion.Tool/Assets/build/Package.targets
+++ b/src/SourceFusion.Tool/Assets/build/Package.targets
@@ -6,12 +6,17 @@
       <!-- 我们使用 $(MSBuildRuntimeType) 来判断编译器是 .NET Core 的还是 .NET Framework 的。-->
       <UseDotNetCore Condition=" '$(MSBuildRuntimeType)' == 'Core'">True</UseDotNetCore>
       <UseDotNetCore Condition=" '$(MSBuildRuntimeType)' != 'Core'">False</UseDotNetCore>
+      <!-- 使用 RuntimeInformation 得到 MSBuild 此刻正在运行的 .NET 框架版本号。-->
+      <MSBuildRuntimeVersion Condition=" '$(UseDotNetCore)' == 'True' ">$([System.Runtime.InteropServices.RuntimeInformation]::FrameworkDescription.ToString())</MSBuildRuntimeVersion>
+      <_TelescopeMSBuildRuntimeFramework Condition=" $(MSBuildRuntimeVersion.StartsWith('.NET Core 3')) ">netcoreapp3.1</_TelescopeMSBuildRuntimeFramework>
+      <_TelescopeMSBuildRuntimeFramework Condition=" $(MSBuildRuntimeVersion.StartsWith('.NET 6')) ">net6.0</_TelescopeMSBuildRuntimeFramework>
+      <_TelescopeMSBuildRuntimeFramework Condition=" '$(_TelescopeMSBuildRuntimeFramework)' == '' ">net5.0</_TelescopeMSBuildRuntimeFramework>
     </PropertyGroup>
     <PropertyGroup Condition=" $(IsInDemoToolDebugMode) != 'True' ">
       <TransformCodeToolCommand>&quot;$(MSBuildThisFileDirectory)../tools/net48/codet.exe&quot;</TransformCodeToolCommand>
     </PropertyGroup>
     <PropertyGroup Condition=" $(UseDotNetCore) == 'True' ">
-      <TransformCodeToolCommand>dotnet &quot;$(MSBuildThisFileDirectory)../tools/netcoreapp3.1/codet.dll&quot;</TransformCodeToolCommand>
+      <TransformCodeToolCommand>dotnet &quot;$(MSBuildThisFileDirectory)../tools/$(_TelescopeMSBuildRuntimeFramework)/codet.dll&quot;</TransformCodeToolCommand>
     </PropertyGroup>
     <PropertyGroup>
       <_SourceFusionDefaultWorkingFolder Condition="'$(_SourceFusionDefaultWorkingFolder)' == ''">obj/$(Configuration)/</_SourceFusionDefaultWorkingFolder>

--- a/src/SourceFusion.Tool/SourceFusion.Tool.csproj
+++ b/src/SourceFusion.Tool/SourceFusion.Tool.csproj
@@ -2,14 +2,14 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>netcoreapp3.1;net48</TargetFrameworks>
+    <TargetFrameworks>net6.0;net5.0;netcoreapp3.1;net48</TargetFrameworks>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <AssemblyName>codet</AssemblyName>
     <RootNamespace>dotnetCampus.SourceFusion</RootNamespace>
     <Product>dotnetCampus.SourceFusion</Product>
     <PackageId>dotnetCampus.SourceFusion</PackageId>
     <!--<GeneratePackageOnBuild>true</GeneratePackageOnBuild>-->
-    <Version>0.9.6-alpha</Version>
+    <Version>0.10.6-alpha</Version>
     <BuildOutputTargetFolder>tools</BuildOutputTargetFolder>
     <Description>
       使代码在编译期执行，以提升运行时效率。

--- a/src/dotnetCampus.Telescope.NuGet/Assets/build/Package.targets
+++ b/src/dotnetCampus.Telescope.NuGet/Assets/build/Package.targets
@@ -6,12 +6,17 @@
       <!-- 我们使用 $(MSBuildRuntimeType) 来判断编译器是 .NET Core 的还是 .NET Framework 的。-->
       <UseDotNetCore Condition=" '$(MSBuildRuntimeType)' == 'Core'">True</UseDotNetCore>
       <UseDotNetCore Condition=" '$(MSBuildRuntimeType)' != 'Core'">False</UseDotNetCore>
+      <!-- 使用 RuntimeInformation 得到 MSBuild 此刻正在运行的 .NET 框架版本号。-->
+      <MSBuildRuntimeVersion Condition=" '$(UseDotNetCore)' == 'True' ">$([System.Runtime.InteropServices.RuntimeInformation]::FrameworkDescription.ToString())</MSBuildRuntimeVersion>
+      <_TelescopeMSBuildRuntimeFramework Condition=" $(MSBuildRuntimeVersion.StartsWith('.NET Core 3')) ">netcoreapp3.1</_TelescopeMSBuildRuntimeFramework>
+      <_TelescopeMSBuildRuntimeFramework Condition=" $(MSBuildRuntimeVersion.StartsWith('.NET 6')) ">net6.0</_TelescopeMSBuildRuntimeFramework>
+      <_TelescopeMSBuildRuntimeFramework Condition=" '$(_TelescopeMSBuildRuntimeFramework)' == '' ">net5.0</_TelescopeMSBuildRuntimeFramework>
     </PropertyGroup>
     <PropertyGroup Condition=" $(IsInDemoToolDebugMode) != 'True' ">
       <TransformCodeToolCommand>&quot;$(MSBuildThisFileDirectory)../tools/net48/dotnetCampus.TelescopeTask.exe&quot;</TransformCodeToolCommand>
     </PropertyGroup>
     <PropertyGroup Condition=" $(UseDotNetCore) == 'True' ">
-      <TransformCodeToolCommand>dotnet &quot;$(MSBuildThisFileDirectory)../tools/netcoreapp3.1/dotnetCampus.TelescopeTask.dll&quot;</TransformCodeToolCommand>
+      <TransformCodeToolCommand>dotnet &quot;$(MSBuildThisFileDirectory)../tools/$(_TelescopeMSBuildRuntimeFramework)/dotnetCampus.TelescopeTask.dll&quot;</TransformCodeToolCommand>
     </PropertyGroup>
     <PropertyGroup>
       <_TelescopeSourceDefaultWorkingFolder Condition="'$(_TelescopeSourceDefaultWorkingFolder)' == ''">obj/$(Configuration)/$(TargetFramework)/</_TelescopeSourceDefaultWorkingFolder>

--- a/src/dotnetCampus.Telescope.NuGet/dotnetCampus.Telescope.NuGet.csproj
+++ b/src/dotnetCampus.Telescope.NuGet/dotnetCampus.Telescope.NuGet.csproj
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp3.1;netstandard2.0;net45</TargetFrameworks>
+    <TargetFrameworks>net6.0;net5.0;netcoreapp3.1;netstandard2.0;net45</TargetFrameworks>
     <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
     <PackageLicenseExpression>MIT</PackageLicenseExpression>
     <PackageId>dotnetCampus.TelescopeSource</PackageId>

--- a/src/dotnetCampus.Telescope.Sample/Directory.Build.targets
+++ b/src/dotnetCampus.Telescope.Sample/Directory.Build.targets
@@ -1,3 +1,3 @@
 ﻿<Project>
-  <!--<Import Project="..\dotnetCampus.Telescope.NuGet\Assets\build\Package.targets" />-->
+  <Import Project="..\dotnetCampus.Telescope.NuGet\Assets\build\Package.targets" />
 </Project>

--- a/src/dotnetCampus.Telescope/dotnetCampus.Telescope.csproj
+++ b/src/dotnetCampus.Telescope/dotnetCampus.Telescope.csproj
@@ -1,7 +1,7 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp3.1;netstandard2.0;net45</TargetFrameworks>
+    <TargetFrameworks>net6.0;net5.0;netcoreapp3.1;netstandard2.0;net45</TargetFrameworks>
     <LangVersion>latest</LangVersion>
     <Nullable>enable</Nullable>
   </PropertyGroup>

--- a/src/dotnetCampus.TelescopeTask/dotnetCampus.TelescopeTask.csproj
+++ b/src/dotnetCampus.TelescopeTask/dotnetCampus.TelescopeTask.csproj
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>netcoreapp3.1;net48</TargetFrameworks>
+    <TargetFrameworks>net6.0;net5.0;netcoreapp3.1;net48</TargetFrameworks>
   </PropertyGroup>
 
   <ItemGroup>

